helm/prometheus-rules/templates/alerting-rules/inhibit.prometheus-agent.rules.yml

```diff
@@ -0,0 +1,30 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  creationTimestamp: null
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: inhibit.prometheus-agent.rules
+  namespace: {{ .Values.namespace  }}
+spec:
+  groups:
+  - name: inhibit.prometheus-agent
+    rules:
+    - alert: InhibitionVintageClusterReleaseBeforeV19
+      annotations:
+        description: '{{`Cluster ({{ $labels.cluster_id }}) using Release ({{ $labels.release_version }}).`}}'
+      expr: aggregation:giantswarm:cluster_release_version{release_version!~"^(1[9]|[2-9]\\d+|[1-9]\\d{2,})\\.[0-9]+\\.[0-9]+.*$"}
+      labels:
+        vintage_cluster_release_before_v19: "true"
+        area: empowerment
+        team: atlas
+        topic: monitoring
+    - alert: InhibitionCAPIClusterDefaultApps
+      annotations:
+        description: '{{`Cluster ({{ $labels.cluster_id }}) using ({{ $labels.app }}) ({{ $labels.version }}).`}}'
+      expr: label_replace(app_operator_app_info{app="observability-bundle",status="deployed",name=~".+-observability-bundle"} , "cluster_id", "$1", "name", "(.+)-observability-bundle")
```

Comment:
Going with the suggestion above, we should probably use absent? What do you think?

I'd rather we inhibit on not running agent

Reply:
that's the right one :D

Reply:
I will test it

Reply:
Can you also ensure it's empty on non capi clusters?

Reply:
Did you test the inhibiton on vintage clusters?

Reply:
tested on vintage

Reply:
using absent is not ideal!

Reply:
we will not having labels

Reply:
we need the cluster_id

Reply:
Does it inhibit vintage clusters then?

Reply:
If not I'm fine :)

Reply:
Well, How to inhibit this CAPI clusters?

Reply:
this inhibit above fires when the app is deployed

Reply:
if we check absent, we lose labels

Reply:
It should fire when the app is deployed :) But I'm afraid this would also fire for vintage clusters when it should not :)

Reply:
if we just remove the vintage inhibition, could be a solution

Reply:
then maybe we do not need the other one :)

Reply:
or checking the namespace pattern.

Reply:
If we can make it work everywhere with one inhibition let's do that :)

Reply:
But how to make the query above fires when the app is not deployed

Reply:
oh right

Reply:
how about using a rate function? rate(label_replace(app_operator_app_info{app="observability-bundle",status="deployed",name=~".+-observability-bundle"} , "cluster_id", "$1", "name", "(.+)-observability-bundle")[15min]) by (cluster_id) > 0 ? Could that work?

Reply:
> 0 means it's deployed?

Reply:
>0 means its deployed yes

Reply:
but maybe it's missing tbh

Reply:
I;m nowhere near a promethues

Reply:
Let's discuss that tomorrow :(

Reply:
I found this, which gets the list of clusterIds and check against our metric.

```
count(label_replace(kube_statefulset_created{namespace=~".+-prometheus"}, "cluster_id", "$1", "namespace", "(.+)-prometheus") ) by (cluster_id) unless count(label_replace(app_operator_app_info{app="observability-bundle",status="deployed",name=~".+-observability-bundle"},"cluster_id", "$1", "name", "(.+)-observability-bundle")) by (cluster_id)
```

Reply:
and it works on vintage also

Reply:
maybe we can remove the other one! and this should work on both